backend/app/api/managers.py

```diff
@@ -0,0 +1,32 @@
+from typing import List
+
+from fastapi import WebSocket
+
+from app.config import settings
+
+
+class WsConnectionManager:
+    def __init__(self) -> None:
+        self.active_connections: dict[int, List[WebSocket]] = {}
+
+    async def connect(self, websocket: WebSocket, room_id: int) -> None:
+        if (
+            self.active_connections.get(room_id)
+            and len(self.active_connections[room_id]) == settings.max_ws_connections
+        ):
+            await websocket.close(reason='Room is full')
```

Comment:
That's a really complicated way to store stuff. This is probably fine as an intermediate solution while the thing that distinguishes a volunteer from a cclient is not yet implemented, but notice that you always want exactly two websockets (moreover, they are not completely interchangeable: the client's websocket logically differs from the volunteer's), so you could just store them as a `Tuple[Optional[WebSocket], Optional[WebSocket]]`

Reply:
I didn't use a tuple since a tuple is an unchangeable data structure, and it will not possible to reconnect then

Reply:
I don't think, this really is an issue as long as the number of connections is constant (you could always create a new tuple with one value updated). However, you mentioned below that the number of connections could vary, and if we want to support this model, the tuple is, of course, not an appropriate data structure.